pkg/tools/tools.go

```diff
@@ -180,3 +183,49 @@ func ToolResultToMap(result any) (map[string]any, error) {
 	}
 	return m, nil
 }
+
+// LoadAndRegisterCustomTools loads tool configurations from a YAML file
+// and registers them.
+func LoadAndRegisterCustomTools() error {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("failed to get user home directory: %w", err)
+	}
+	// Custom tools are stored in ~/.config/kubectl-ai/tools.yaml
+	configPath := filepath.Join(home, ".config", "kubectl-ai", "tools.yaml")
```

Comment:
This has the potential to become quite large. Could each tool be a separate file in a tools directory like `$HOME/.config/kubectl-ai/tools/helm.yaml`?

Reply:
I discussed this in a comment thread. pl. take a look. My feedback is driven more by the UX.